lib/tests/test_merged_tree.rs

```diff
@@ -1182,3 +1183,136 @@ fn test_merge_simplify_result() {
+  rewriting commits on the `main` branch.
+
+* `jj op log` now supports `--no-graph`.
+{suffix}"#
+    );
+    let expected_text = format!(
+        r#"{prefix}
+* You can now configure the set of immutable commits via
+  `revsets.immutable-heads`. For example, set it to `"main"` to prevent
+  rewriting commits on the `main` branch. The new `immutable()` revset resolves
+  to these immutable commits.
+
+* `jj op log` now supports `--no-graph`.
+{suffix}"#
+    );
+
+    // conflict in parent commit
+    let parent_base = create_single_tree(repo, &[(&conflict_path, &parent_base_text)]);
+    let parent_left = create_single_tree(repo, &[(&conflict_path, &parent_left_text)]);
+    let parent_right = create_single_tree(repo, &[(&conflict_path, &parent_right_text)]);
+    let parent_merged = MergedTree::new(Merge::new(
+        vec![parent_base],
+        vec![parent_left, parent_right],
+    ));
+
+    // different conflict in child
+    let child1_base = create_single_tree(
+        repo,
+        &[(&other_path, "child1"), (&conflict_path, &parent_base_text)],
+    );
+    let child1_left = create_single_tree(
+        repo,
+        &[(&other_path, "child1"), (&conflict_path, &parent_left_text)],
+    );
+    let child1_right = create_single_tree(
+        repo,
+        &[
+            (&other_path, "child1"),
+            (&conflict_path, &child1_right_text),
+        ],
+    );
+    let child1_merged = MergedTree::new(Merge::new(
+        vec![child1_base],
+        vec![child1_left, child1_right],
+    ));
+
+    // resolved state
+    let child2 = create_single_tree(repo, &[(&conflict_path, &child2_text)]);
+    let child2_merged = MergedTree::resolved(child2.clone());
+
+    // expected result
+    let expected = create_single_tree(
+        repo,
+        &[(&other_path, "child1"), (&conflict_path, &expected_text)],
+    );
+    let expected_merged = MergedTree::resolved(expected);
+
+    let merged = child1_merged.merge(&parent_merged, &child2_merged).unwrap();
+    assert_eq!(merged, expected_merged);
+
+    // Also test the setup by checking that the unsimplified content conflict cannot
+    // be resolved. If we later change files::merge() so this no longer fails,  it
+    // probably means that we can delete this whole test (the Merge::simplify() call
+    // in try_resolve_file_conflict() is just an optimization then).
+    let text_merge = Merge::new(
+        vec![Merge::new(
+            vec![parent_base_text.as_bytes()],
+            vec![parent_left_text.as_bytes(), parent_right_text.as_bytes()],
+        )],
+        vec![
+            Merge::new(
+                vec![parent_base_text.as_bytes()],
+                vec![parent_left_text.as_bytes(), child1_right_text.as_bytes()],
+            ),
+            Merge::resolved(child2_text.as_bytes()),
+        ],
+    );
+    assert!(matches!(
+        jj_lib::files::merge(text_merge.flatten()),
+        MergeResult::Conflict(_)
+    ));
```

Comment:
IIUC, `jj status` saw this seven-way conflict. How did it happen that, in your working copy, there were no conflict markers?

Is it the case that `text_merge.simplify()` (or something like that) resolves the conflict? If so, you could show this in the test.

(this is something I could probably figure out myself, but it seems quicker to ask whether I'm on the right track).

Reply:
Maybe this was a silly question and I just need to carefully follow what MergedTree::merge does.

Reply:
> IIUC, `jj status` saw this seven-way conflict. How did it happen that, in your working copy, there were no conflict markers?

No, the 7-way conflict was only seen transiently by the merge code. Because of the change in `other_path`, the whole tree cannot be simplified to a 3-way conflict by `.flatten().simplify()` in `MergedTree::merge()`. Instead, the 7-way conflict gets passed into `files::merge()`, which fails, and propagates up to the root tree. The trivial merge in `other_path` has been applied, however, so the 7-way merge in the root tree can now be simplified to a 3-way conflict by the final `.simplify()` in `MergedTree::merge()`. That gets written to the commit.

Then, when that simpler 3-way conflict is later checked out the working copy, we materialize the conflict there, but since the 3-way conflict is simpler than the 7-way conflict, it can be automatically resolved, so no conflict markers actually get materalized.